### ISSUE TITLE
Diagnostic Sensors not created for all equipment: [Discharge Air Temperature is reporting Open instead of the actual temperature]

### ISSUE THREAD

**Issue by subtlybeast** [PeteRager/lennoxs30#181]
Integration Version 0.2.10

Within the "Indoor Unit", the Sensor "System_iu_Discharge_Air_Temperature" reports "Open" when it should report a numeric value; Temperature, as measured from the Discharge Air Sensor, in the Temperature Unit that is set in the S30.

**Comment by PeteRager** [comment]
Thanks for the note, from what I can tell this is just what it reports.  Here is the data from my system:

```
                        {
                            "diagnostic": {
                                "name": "Discharge Air Temperature",
                                "unit": "",
                                "value": "open"
                            },
                            "id": 7
                        },
```

To get the data from your system, follow the instructions for enabling message logging to a file in the readme.  
https://github.com/PeteRager/lennoxs30#reporting-bugs

Restart HA or Reload the integration, wait a minute or two.  Then look in the file for this message and/or post the file here.

In this context I believe the word "open" means there is no sensor installed.  The other uses of "Open" (note the capitalized O), seem to indicate the switch, etc, is actually open.

When you say " as measured from the Discharge Air Sensor, in the Temperature Unit that is set in the S30.", do you see data in the S30 display that shows this temperature?  If so it'd be good to know that value, perhaps we can then find it in the message log?

**Comment by PeteRager** [comment]
Super.  Can you post your logfile here or email to me at pete dot sage at icloud dot com

I'll try to figure out why it isn't showing up.

**Comment by subtlybeast** [comment]
Here goes
[S30_Local.log](https://github.com/PeteRager/lennoxs30/files/9095218/S30_Local.log)

Thanks

**Comment by PeteRager** [comment]
Very cool.   So your system have a zoning controller which is equipment Id 3.  Right now the integration is not creating a device for it.  That’ll need to be fixed.  

However, it should be attempting to create a sensor.  

It’s name should be System_3_Discharge_Air_Temperature

See if you can find that.  

It may also be logging a warning in the home assistant log

        _LOGGER.warning(
            f"Unexpected equipment id [{self.eid}], please raise an issue and post a mesage log"
        )
        return {
            "identifiers": {(DOMAIN, self._system.unique_id())},
        }

But it’s also possible that whole path is broken because it’s never been tested.   I’ll load your file into the simulator next day or so and see what comes up.

**Comment by subtlybeast** [comment]
Makes sense, the Discharge Air Temperature appears under the Zoning Controller section on the S30 diagnostics screen, that might be why it is reading open now as the zoning controller is taking over it in my configuration, and the device is not created in the integration at this point.

in the home assistant log there is the
"Unexpected equipment id [{self.eid}], please raise an issue and post a mesage log"

This is cool - thank you :)

**Comment by PeteRager** [comment]
I've made some initial progress.  Probably a week or so to get the changes done, maybe less.  I'll keep you posted when I have something to share.

**Comment by PeteRager** [comment]
From what I can tell the second sensor should be getting created.

if you got into developer tools / states, do you see something similar to this?

sensor.System_3_Discharge_Air_Temperature

**Comment by subtlybeast** [comment]
The only one I see under developer tools / states is 
sensor.system_iu_discharge_air_temperature

**Comment by PeteRager** [comment]
Got the changes working, next step is testing and packaging for release.  Here is what your system looks like with the changes.

Here is the device list.  Note the zoning controller is showing up and the integration is pulling in the correct model numbers for the installed equipment.

![devices](https://user-images.githubusercontent.com/76050312/179513267-dad40714-915a-4dc0-8ac0-1c2239acf8a9.PNG)

Here is the view of the zoning controller device.  The diag sensors are showing up now and the serial number of the zoning controller is showing up under the device Hardware attribute (I changed the digits from what your serial number is)

![zoning_device](https://user-images.githubusercontent.com/76050312/179513504-c86a284d-2d4a-48c4-8254-c6c092a8f9be.PNG)

Lastly here's a dashboard to more clearly show the new attributes.

![diag_sensors](https://user-images.githubusercontent.com/76050312/179513752-1812b032-9a43-426f-b160-9beca3503e4c.PNG)

**Comment by subtlybeast** [comment]
NICE!
Thank you very much! Now I will be able to properly collect and see data points and hopefully tune that system

**Comment by PeteRager** [comment]
Merging pull request, auto-closed.  Creating release

**Comment by PeteRager** [comment]
Release is here.  Let me know how it goes.

https://github.com/PeteRager/lennoxs30/releases/tag/0.2.11
